### DISCUSSION TITLE
[V2] gatsby-plugin-layout and material-ui conflicting

### DISCUSSION THREAD

**Discussion by MSFTserver** [gatsbyjs/gatsby#35679] [Help] [answered]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

I recently tried to compile a website i made around a year ago with gatsby v2 and node v12 using gatsby-plugin-layout and material-ui but for some strange reason even if i set all versions for gatsby, react, material-ui and the other theming functions the same as when it did work, my tags don't work any more across the site when another page uses material-ui for its theming

<details>
  <summary>[Hidden] Relevant Code</summary>

### a little back story on where material-ui is coming from
this snippet gets injected from chonky.io as `<FileBrowser>`

```js 
   // chonky/src/components/external/FileBrowser.tsx
   // https://github.com/TimboKZ/Chonky/blob/2.x-dev/packages/chonky/src/components/external/FileBrowser.tsx
    return (
        <IntlProvider locale="en" defaultLocale="en" {...i18n}>
            <ChonkyFormattersContext.Provider value={formatters}>
                <ReduxProvider store={store}>
                    <ThemeProvider theme={theme}>
                        <MuiThemeProvider theme={theme}>
                            <ChonkyIconContext.Provider
                                value={
                                    iconComponent ??
                                    defaultConfig.iconComponent ??
                                    ChonkyIconPlaceholder
                                }
                            >
                                {disableDragAndDrop || disableDragAndDropProvider ? (
                                    chonkyComps
                                ) : (
                                    <DndProvider backend={HTML5Backend}>
                                        {chonkyComps}
                                    </DndProvider>
                                )}
                            </ChonkyIconContext.Provider>
                        </MuiThemeProvider>
                    </ThemeProvider>
                </ReduxProvider>
            </ChonkyFormattersContext.Provider>
        </IntlProvider>
    );
```

this is where the material-ui from chonky makes its way into my project

```js
 // static/ChonkyBrowse.js
 // https://github.com/MSFTserver/print2a.com/blob/main/static/ChonkyBrowse.js
  return (
      <FileBrowser
        files={currentNodes}
        folderChain={folderChain}
        onFileAction={handleFileOpen}
        fileActions={[
          ChonkyActions.DownloadFiles
        ]}
        disableDragAndDrop={true}
        darkMode={true}
        //view={FileView.SmallThumbs}
      >
        <FileNavbar />
        <FileToolbar />
        <FileList />
        <FileContextMenu />
      </FileBrowser>
  );
```

This snippet above gets injected under the `<App>` tag as a child

```js
// layouts/Template/Template.js
// https://github.com/MSFTserver/print2a.com/blob/main/src/layouts/Template/Template.js
return (
    <Layout {...layout}>
      <Background
        {...background}
        animation={{ show, ...background.animation }}
      >
        {<App>{children}</App>}
      </Background>
    </Layout>
);
```
  
</details>

### Reproduction Link

https://github.com/MSFTserver/print2a.com

### Steps to Reproduce
Use node v12
1. `npm install`
2. `npm run develop`
3. view site at `localhost:14000`

### Expected Result

`<Layout>` tags and material-ui themeProvider should work together as they used to back in my old project

![image](https://user-images.githubusercontent.com/18357967/168354800-23a37010-8aef-4114-832e-e4a19b04c953.png)

### Actual Result

material-ui theming is taking over the `<Layout>` tag and not actually using it anymore for any of my pages. this would be controlling the main body and HTML of the "default" theme across pages

![image](https://user-images.githubusercontent.com/18357967/168357698-1a08dc99-4f90-4703-9948-5ca193024e13.png)

### Environment

```shell
System:
    OS: Windows 10 10.0.19044
    CPU: (8) x64 Intel(R) Core(TM) i7-6700K CPU @ 4.00GHz
  Binaries:
    Node: 12.22.12 - C:\Program Files\nodejs\node.EXE
    npm: 6.14.16 - C:\Program Files\nodejs\npm.CMD
  Languages:
    Python: 3.8.10 - C:\Program Files\Python38\python.EXE
  Browsers:
    Chrome: 101.0.4951.64
    Edge: Spartan (44.19041.1266.0), Chromium (101.0.1210.39)
  npmPackages:
    gatsby: 2.30.1 => 2.30.1
    gatsby-plugin-google-analytics: ^2.0.18 => 2.11.0
    gatsby-plugin-jss: 2.0.9 => 2.0.9
    gatsby-plugin-layout: ^1.0.0 => 1.10.0
    gatsby-plugin-manifest: ^2.10.0 => 2.12.1
    gatsby-plugin-page-creator: 2.8.0 => 2.8.0
    gatsby-plugin-react-helmet: 3.0.12 => 3.0.12
    gatsby-plugin-sitemap: ^2.1.0 => 2.12.0
```

**Comment by MSFTserver** [comment]
Atleast trying to fix this so i can migrate it to the newer versions of gatsby and other packages, or if theres an easy way i can migrate from Layout v1 to material-ui

**Comment by HKAgrwl** [comment]
Can you assign this issue to me?

**Comment by MSFTserver** [comment]
> Can you assign this issue to me?

Unfortunately a maintainer would have to assign you, i would if i could

**Comment by pieh** [comment]
I just pulled reproduction and tried to run locally (with `develop` as that's what was mentioned in original post) and it seems that now things are working:

![Screenshot 2022-05-16 at 14 45 24](https://user-images.githubusercontent.com/419821/168595480-483ed32b-d6f2-4757-94fb-c97dbdc7dde8.png)
 
Did you manage to fix it yourself? Or did you migrate to something else completely? (or maybe there is other reason why it works for me?)

**Comment by MSFTserver** [comment]
@pieh yea sorry i literally just updated the links, i got my dev repo working by upgrading all packages but for some reason upon doing that i loose the theme object that is suppose to be added to props, this used to be the case but isnt anymore, i did a dirty work around for loosing access to this.props.theme and added `const theme = require('../../settings/theme');` to every single file that needed to access it. While that is issue when i upgrade there is still the issue of older version stuff also not working here https://github.com/MSFTserver/print2a.com

_Sent from my SM-G988U using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Reply by MSFTserver** [reply]
ok scratch that it works on develop mode with all packages updated but upon build method all styling is broken on the initial load of the static site but if you navigate pages styling comes back

**Comment by MSFTserver** [comment, marked as answer]
I decided to just dump this old project and migrate away from gatsby
